lib/simpler/router/route.rb

```diff
@@ -12,9 +12,30 @@ def initialize(method, path, controller, action)
       end
 
       def match?(method, path)
-        @method == method && path.match(@path)
+        @method == method && path_match?(path)
       end
 
+      def variables(path)
+        path_request = split_path(path)
+        path_route = split_path(@path)
+        variables = {}
+        path_route.zip(path_request).each { |route, request| variables[route[1..-1].to_sym] = request if route[0] == ':' }
+        variables
+      end
+
+      private
+
+      def path_match?(path)
+        path_request = split_path(path)
+        path_route = split_path(@path)
+        return false if path_request.count != path_route.count
+        path_route.zip(path_request).each { |route, request| return false if route[0] != ':' && request != route }
```

Comment:
Если нужно проверить, что все элементы коллекции удовлетворяют некоторому условию, используй `all?`
Но тут вообще странность - похоже, что роут без параметров (`get '/tests'`) никогда не сматчится

Reply:
В общем, вот такой вариант читабельнее.
```suggestion
        path_route.zip(path_request).all? { |route, request| request == route || route[0] == ':' }
```
В модуле Enumerable очень много полезных методов есть - изучи, это главный инструмент рубиста